Library/Homebrew/cmd/update-report.rb

```diff
@@ -249,8 +249,21 @@ def updated?
   def migrate_tap_migration
     report[:D].each do |full_name|
       name = full_name.split("/").last
-      next unless (dir = HOMEBREW_CELLAR/name).exist? # skip if formula is not installed.
       next unless new_tap_name = tap.tap_migrations[name] # skip if formula is not in tap_migrations list.
+      if tap == "caskroom/cask"
+        next unless (HOMEBREW_REPOSITORY/"Caskroom"/name).exist?
+        new_tap = Tap.fetch(new_tap_name)
+        new_tap.install unless new_tap.installed?
+        system HOMEBREW_BREW_FILE, "install", name
+        system HOMEBREW_BREW_FILE, "link", name, "--force"
```

Comment:
I'm not sure if this is a problem of practical relevance, but it feels like it could be safer to use `full_name` instead of just `name` in case a Cask is ever migrated to a non-core formula and there is some ambiguity. But I have to admit that I haven't fully though this through, thus feel free to educate me if that's not going to happen or not an issue …